lxd-agent/api_1.0.go

```diff
@@ -61,3 +67,89 @@ func api10Get(d *Daemon, r *http.Request) response.Response {
 
 	return response.SyncResponseETag(true, fullSrv, fullSrv)
 }
+
+func api10Put(d *Daemon, r *http.Request) response.Response {
+	var data api.API10Put
+
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		return response.ErrorResponse(http.StatusInternalServerError, err.Error())
+	}
+
+	d.serverCID = data.CID
```

Comment:
We'll want a mutex around these vars to ensure we don't race accessing them when they are being set.